gems/aws-sdk-s3/lib/aws-sdk-s3/multipart_file_uploader.rb

```diff
@@ -175,7 +175,6 @@ def upload_in_threads(pending, completed, options)
               error
             end
           end
-          thread.abort_on_exception = true
```

Comment:
I was tracking down how errors are raised for both `multipart_file_uploader` and `multipart_stream_uploader`. Below is a breakdown for `multipart_file_uploader` but the stream uploader is pretty similar:
* Errors are rescued instead of raised: https://github.com/aws/aws-sdk-ruby/blob/2f0e6f83473c59d6f4875694ae8cb386a9894a61/gems/aws-sdk-s3/lib/aws-sdk-s3/multipart_file_uploader.rb#L172
* All the errors are collected to `threads_errors` and then passed to abort multipart upload call: 
  * https://github.com/aws/aws-sdk-ruby/blob/2f0e6f83473c59d6f4875694ae8cb386a9894a61/gems/aws-sdk-s3/lib/aws-sdk-s3/multipart_file_uploader.rb#L79 
  * https://github.com/aws/aws-sdk-ruby/blob/2f0e6f83473c59d6f4875694ae8cb386a9894a61/gems/aws-sdk-s3/lib/aws-sdk-s3/multipart_file_uploader.rb#L86

Removing `abort_on_exception` should be _ok_ here.

The `file_downloader.rb` is bit different since we raise the error in its own thread. With the removal of `abort_on_exception`, we shouldn't cause any interruption in the main thread.